### PR TITLE
Add details on operational mitigations

### LINKED ISSUE #33: Make sure mitigations are documented properly
For the service threat model: document thoroughly all the operational mitigations and the risks and limitations due to unmitigations.

Give recommendations and examples around how to set up the environment to provide sufficient security, including:
- Security Module rules (e.g. SELinux, SMACK) that could be used to protect access to the data stores, IPC communications and hardware providers
- Minimal Capabilities required, seccomp rules, etc. for the PARSEC service process
- Recommendations for use of encrypted file systems

### PR DESCRIPTION
**Pull request by hug-dev**
Fix #33 (remaining items)

### CODE REVIEW

File: src/threat_model/threat_model.md [hunk 1]
```diff
@@ -324,18 +324,21 @@ cleared.
...
+| 3    | Mappings should be put in a location that only the Parsec service and the trusted administrator can access.                                                        | The mappings are inside the `parsec` user home folder.                                                                                                   |
+| 4    | The trusted administrator needs to be notified when any kind of resource is running out.                                                                           | Currently only the logs can be used as a way of knowing that the system is failing because a resource is running out.                                    |
+| 5    | Parsec logs coming from the service binary should be redirected to a file that is only writable by the service and readable by the trusted administrator.          | Using systemd, the logs can only be written by the service and read by the `parsec` user. Otherwise, they can be stored in the `parsec` home folder.     |
```
Comment (ionut-arm) [inline src/threat_model/threat_model.md:337]: Are you sure the part about "can only be read by the `parsec` user" is factual? On my machine, at least, everyone in the `adm` group can read them.

File: src/threat_model/threat_model.md [hunk 2]
```diff
@@ -324,18 +324,21 @@ cleared.
...
+| 6    | Parsec configuration file should be only writable by the trusted administrator and readable by the service.                                                        | The configuration is in the `parsec` home folder.                                                                                                        |
+| 7    | The trusted administrator needs to check that during the boot process the trusted identity provider has successfully given the root trust bundle to the service.   | This is a manual check. It is possible that Parsec would not initiliaze correctly if the root trust bundle has not been given to it.                     |
```
Comment (ionut-arm) [inline src/threat_model/threat_model.md:339]:
```suggestion
| 7    | The trusted administrator needs to check that during the boot process the trusted identity provider has successfully given the root trust bundle to the service.   | This is a manual check. It is possible that Parsec would not initialize correctly if the root trust bundle has not been given to it.                     |
```

File: src/threat_model/threat_model.md [hunk 3]
```diff
@@ -324,18 +324,21 @@ cleared.
...
+| 8    | The hardware descriptors should only be accessible by trusted privileged processes.                                                                                | Manual check but this should be the default on most systems.                                                                                             |
+| 9    | The Listener endpoint should be contained in a location that only the Parsec service and the trusted administrator can access (only they can create the endpoint). | The endpoint folder is owned by the `parsec` user and only writable by it.                                                                               |
+| *10* | *A set of mutually trusted clients has restricted read-write access to the service IPC endpoint.*                                                                  | The `parsec-clients` group only has read-access on the socket.                                                                                           |
```
Comment (ionut-arm) [inline src/threat_model/threat_model.md:342]: Should the third column be italic as well?

File: src/threat_model/threat_model.md (hunk 1 above)
Comment (hug-dev) [reply src/threat_model/threat_model.md:337]: I remember I tried it once and I couldn't read the service logs with `journalclt --user -u parsec` when logged as another user.
I guess there are privileged users or groups that can still read them but then this is implied by assumption 4:

> Users with privilege rights are trusted to exercise them in a non-malicious way.

**Review by ionut-arm — APPROVED** [review]

**Pull request merged**